crates/turbopack-core/src/resolve/options.rs

```diff
@@ -362,25 +404,25 @@ impl ImportMap {
         lookup_path: Vc<FileSystemPath>,
         request: Vc<Request>,
     ) -> Result<ImportMapResult> {
-        // TODO lookup pattern
         // relative requests must not match global wildcard aliases.
-        if let Some(request_string) = request.await?.request() {
-            let mut lookup = if request_string.starts_with("./") {
+
+        if let Some(request_pattern) = request.await?.request_pattern() {
+            let mut lookup = if request_pattern.must_match("./") {
```

Comment:
A hard if else won't work here. The pattern could match both paths.

You can use `split_could_match` to split the Pattern into the part the matches `"./"` and the part that doesn't.